src/theme/typography/helper.ts

```diff
@@ -0,0 +1,101 @@
+import { CSSProp, css } from "styled-components";
+import { StyledComponentProps, getCollection, getValue } from "../helpers";
+import { FontScheme, ThemeCollectionSpec } from "../types";
+import { LifeSgTypographySet } from "./specs/typography-set";
+import { TypographyCollectionMap, TypographySet } from "./types";
+
+const TypographySpec: ThemeCollectionSpec<TypographyCollectionMap, FontScheme> =
+    {
+        collections: {
+            lifesg: LifeSgTypographySet,
+            bookingsg: LifeSgTypographySet,
+            rbs: LifeSgTypographySet,
+            mylegacy: LifeSgTypographySet,
+            ccube: LifeSgTypographySet,
+        },
+        defaultValue: "lifesg",
+    };
+
+export const getTypography = (key: keyof TypographySet) => {
+    return (props: StyledComponentProps) => {
+        const theme = props.theme;
+        const typographySet: TypographySet = getCollection(
+            TypographySpec,
+            theme.fontScheme
+        );
+
+        // Check for an override
+        const typographyValue =
+            theme.overrides && theme.overrides.typography
+                ? getValue(typographySet, key, theme.overrides.typography)
+                : typographySet[key];
+
+        // If function, resolve with props
+        if (typeof typographyValue === "function") {
+            return css`
+                ${(typographyValue as (props: any) => string)(props)}
+            `;
```

Comment:
no need to nest in `css`. since the value is supposed to be CSSProp or string, you should be able to return `typographyValue` without having to invoke it